src/main/java/org/symphonyoss/symphony/messageml/elements/DateSelector.java

```diff
@@ -13,29 +20,110 @@
+        super.buildAll(context, element);
+        break;
+      case PRESENTATIONML:
+        buildElementFromPresentationML(context, element);
+        this.validate();
+        break;
+      default:
+        throw new InvalidInputException(String.format("Invalid message format for \"%s\" element", MESSAGEML_TAG));
+    }
+  }
+
   @Override
   public void validate() throws InvalidInputException {
     super.validate();
-    assertNoAttributes();
+    
+    if (getAttribute(NAME_ATTR) == null) {
+      throw new InvalidInputException("The attribute \"name\" is required");
+    }
+
     assertNoContent();
   }
 
   @Override
   public void asPresentationML(XmlPrintStream out) {
-    out.printElement(PRESENTATIONML_TAG, null, CLASS_ATTR, MESSAGEML_TAG);
+    Map<String, String> presentationAttrs = buildDateSelectorInputAttributes();
+    out.printElement(PRESENTATIONML_TAG, presentationAttrs);
   }
 
   @Override
   public Node asMarkdown() {
-    return new FormElementNode(MARKDOWN);
+    return new FormElementNode(MARKDOWN, getAttribute(NAME_ATTR));
+  }
+
+  @Override
+  protected void buildAttribute(org.w3c.dom.Node item) throws InvalidInputException {
+    switch (item.getNodeName()) {
+      case NAME_ATTR:
+        setAttribute(NAME_ATTR, getStringAttribute(item));
+        break;
+      case PLACEHOLDER_ATTR:
+        setAttribute(PLACEHOLDER_ATTR, getStringAttribute(item));
+        break;
+      default:
+        throw new InvalidInputException("Attribute \"" + item.getNodeName()
+            + "\" is not allowed in \"" + getMessageMLTag() + "\"");
+    }
+  }
+
+  private Map<String, String> buildDateSelectorInputAttributes() {
+    Map<String, String> presentationAttrs = new LinkedHashMap<>();
+
+    presentationAttrs.put(CLASS_ATTR, MESSAGEML_TAG);
+    presentationAttrs.put(PRESENTATIONML_NAME_ATTR, getAttribute(NAME_ATTR));
+
+    if(getAttribute(PLACEHOLDER_ATTR) != null) {
+      presentationAttrs.put(PRESENTATIONML_PLACEHOLDER_ATTR, getAttribute(PLACEHOLDER_ATTR));
+    }
+
+    return presentationAttrs;
+  }
+
+  void buildElementFromPresentationML(MessageMLParser context, org.w3c.dom.Element element) throws InvalidInputException, ProcessingException {
+
+    if (!"".equals(element.getAttribute(PRESENTATIONML_NAME_ATTR))) {
+      String nameValue = element.getAttribute(PRESENTATIONML_NAME_ATTR);
+      element.setAttribute(NAME_ATTR, nameValue);
+      element.removeAttribute(PRESENTATIONML_NAME_ATTR);
+    }
+
+    if (!"".equals(element.getAttribute(PRESENTATIONML_PLACEHOLDER_ATTR))) {
+      String placeholderValue = element.getAttribute(PRESENTATIONML_PLACEHOLDER_ATTR);
+      element.setAttribute(PLACEHOLDER_ATTR, placeholderValue);
+      element.removeAttribute(PRESENTATIONML_PLACEHOLDER_ATTR);
+    }
+
+    NamedNodeMap attributes = element.getAttributes();
+
+    for (int i = 0; i < attributes.getLength(); i++) {
+      buildAttribute(attributes.item(i));
+    }
+
+    NodeList children = element.getChildNodes();
```

Comment:
This element has no child nodes, you don't need to do this part.

Reply:
If we want our unit tests with invalid cases containing text children to continue passing, we must have this line here to build the PresentationML children properly.